### ISSUE TITLE
Adding equivalent of Pandas .replace() or Dask .mask() 

### ISSUE THREAD

**Issue by egoolish** [rapidsai/cudf#184]
Would be incredibly helpful to have a built-in equivalent to the .replace() method in Pandas (https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.replace.html). Current workaround for doing so would be using a CUDA JIT function such as:
```
from pygdf import Series
import numpy as np
from numba import cuda

@cuda.jit
def replace_func(io_array, set_series):
    pos = cuda.grid(1)
    if pos < io_array.size:
        for each in set_series:
            if io_array[pos] == each:
                io_array[pos] = -1
                break

df = Series([5, 1, 8, 2, 4, 8, 3, 7, 4, 3, 1, 2])
print(df)
values_to_replace = [2, 4, 6, 8]
replace_func[1, 512](df.to_gpu_array(), Series(values_to_replace).to_gpu_array())
print(df)
```
```
 0    5
 1    1
 2    8
 3    2
 4    4
 5    8
 6    3
 7    7
 8    4
 9    3
10    1
11    2

 0    5
 1    1
 2   -1
 3   -1
 4   -1
 5   -1
 6    3
 7    7
 8   -1
 9    3
10    1
11   -1
```
For single scalar replacements, applymap is also a possibility. Current problem is inelasticity / need to define custom functions for every use case. 

Harder would be a custom dictionary replacement (e.g. 1 is replaced by -1, 5 is replaced by -10, etc), considering nopython support for dictionary type is limited.

**Comment by kkraus14** [comment]
> Harder would be a custom dictionary replacement (e.g. 1 is replaced by -1, 5 is replaced by -10, etc), considering nopython support for dictionary type is limited.

A workaround for this could be something like creating a second GDF with your dictionary of replacement values, joining onto the original dataframe with a left join, and then running an apply_rows UDF that takes the value from the joined column if not null, otherwise take the value from the original column.

**Comment by kkraus14** [comment]
Fixed by #539
